apm-agent-plugins/apm-servlet-plugin/src/main/java/co/elastic/apm/agent/servlet/RequestDispatcherInstrumentation.java

```diff
@@ -0,0 +1,73 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2019 Elastic and contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.servlet;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import javax.annotation.Nullable;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.http.HttpServletRequest;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+public class RequestDispatcherInstrumentation extends ElasticApmInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return RequestDispatcherInstrumentation.RequestDispatcherAdvice.class;
+    }
+
+    public static class RequestDispatcherAdvice extends RequestDispatcherInstrumentation {
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        private static void beforeExecute(@Advice.Argument(0) @Nullable String path,
+                                          @Advice.This @Nullable Object thiz) {
+            if (thiz instanceof HttpServletRequest) {
+                ((HttpServletRequest) thiz).setAttribute(RequestDispatcher.FORWARD_SERVLET_PATH, (path != null) ? path : "");
```

Comment:
Isn't that property set by the servlet container by default? Do we really have to set it explicitly via instrumentation?

Reply:
no, it isn't set by default. I tested on tomcat 8.5.5

Reply:
That would be very surprising. The javadoc for `javax.servlet.RequestDispatcher#FORWARD_SERVLET_PATH` states:
> The name of the request attribute under which the original
servlet path is made available to the target of a 
{@link #forward(ServletRequest,ServletResponse) forward}